Tests/Test_VoltTempMonitorTask.c

```diff
@@ -0,0 +1,162 @@
+/* Copyright (c) 2021 UT Longhorn Racing Solar */
+
+#include "common.h"
+#include "config.h"
+#include "CANbus.h"
+#include "os.h"
+#include "Tasks.h"
+#include "stm32f4xx.h"
+#include "BSP_Lights.h"
+#include "BSP_PLL.h"
+#include "CAN_Queue.h"
+
+/******************************************************************************
+ * VoltTempMonitor Task Test Plan
+ * 
+ * 1. Build the BPS code for the stm32f413 for this test file (see README.md)
+ * 2. Set up the BPS leader, Minion boards, and Contactor.
+ * 3. Flash the BPS (see README.md)
+ * 4. Connect a logic analyzer to the BPS's CAN port
+ * 5. Start recording the BPS's CAN output
+ * 6. Provide UVOLT, SAFE, OVOLT, OWIRE, and OTEMP cases before starting task 
+ * and see if task fails for all cases (hear contactor open, see lights turn on).
+ * 7. Allow task to run and then check for all fault cases again.
+ * 8. Check the logic analyzer recording for the BPS's CAN messages and verfiy
+ *    that the observed CAN messages contain accurate readings.
+ *****************************************************************************/
+
+// Used by Task1
+OS_TCB Task1_TCB;
+CPU_STK Task1_Stk[DEFAULT_STACK_SIZE];
+
+
+// Initialization task for this test
+void Task1(void *p_arg){
+	OS_CPU_SysTickInit(SystemCoreClock / (CPU_INT32U) OSCfg_TickRate_Hz);
+
+    OS_ERR err;
+    
+    OSSemCreate(&Fault_Sem4,
+                "Fault/Tripped Semaphore",
+                0,
+                &err);
+    assertOSError(err);
+
+    OSSemCreate(&SafetyCheck_Sem4,
+                "Safety Check Semaphore",
+                0,
+                &err);
+    assertOSError(err);
+
+    OSMutexCreate(&WDog_Mutex,
+                "Watchdog Mutex",
+                &err);
+    assertOSError(err);
```

Comment:
You will also need a task to periodically set the watchdog bitmap for the amperes task and the battery balancing task